action-scripts/brownie/scripts/pokeInjectors.py

```diff
@@ -3,35 +3,55 @@
 import os
 
 
+CHAINS_TO_KEEP = [
+    "zkevm-main",
+    "gnosis-main",
+    "mode-main",
+    "linea-main",
+    "fraxtal-main",
+]
+
+
 def main():
     ## LOAD wallet
     mnemonic = os.environ["KEYWORDS"]
     account = accounts.from_mnemonic(mnemonic)
     print(f"Keeper Address: {account.address}")
-    ## ZKEVM
-    network.disconnect()
-    network.connect("zkevm-main")
-    book = AddrBook("zkevm")
-    injectors = book.extras.maxiKeepers.gaugeRewardsInjectors.values()
-    for injectorAddress in injectors:
-        injector = Contract(injectorAddress)
-        (ready, performdata) = injector.checkUpkeep(b"")
-        if ready:
-            injector.performUpkeep(performdata, {"from": account})
-        else:
-            print(f"ZKEVM: {injector.address} not ready")
-    ## Gnosis
-    network.disconnect()
-    network.connect("gnosis-main")
-    book = AddrBook("gnosis")
-    injectors = book.extras.maxiKeepers.gaugeRewardsInjectors.values()
-    for injectorAddress in injectors:
-        injector = Contract(injectorAddress)
-        (ready, performdata) = injector.checkUpkeep(b"")
-        if ready:
-            injector.performUpkeep(performdata, {"from": account})
-        else:
-            print(f"GNOSIS: {injector.address} not ready")
+    for chain in CHAINS_TO_KEEP:
+        try:
+            network.disconnect()
+        except:
+            pass
+        try:
+            network.connect(chain)
+        except:
+            print(f"Skipping {chain} because no network can be found")
+            continue
+        try:
+            book = AddrBook(chain.replace("-main", ""))
+        except:
+            print(
+                f"Skipping {chain} because no Address Information can be found in bal_addresses"
+            )
+            continue
+        to_poke = []
+        try:
+            to_poke = book.extras.maxiKeepers.gaugeRewardsInjectors.values()
+        except:
+            print(f"no gaugeRewardsInjectors found in {chain}")
```

Comment:
```suggestion
        try:
            to_poke = book.extras.maxiKeepers.gaugeRewardsInjectors.values()
        except:
            to_poke = []
            print(f"no gaugeRewardsInjectors found in {chain}")
```

Reply:
it is minor, but the idea is that you only need to initialise it if the address book lookup doesnt work

Reply:
way i see it is L40 (`to_poke = book...`) initialises the `to_poke` var. if that action is somehow caught, then `to_poke` needs to be set to `[]`.

your current solution initialises `to_poke` to `[]`, even though the majority of times it will then be reinitialised to `book...`. doesnt make any sense

im suggesting to _try_ to set `to_poke` to `book...`, but if that doesnt work set it to an empty list instead

Reply:
The way I see it, the script could some day be extended to have additional conditions that could add to the list.  By setting it to empty in any of the conditions you make that less possible.

My general pattern for building lists tends to be initialize it empty and then add to it if there is a reason,a nd if nothing adds to it you have an empty list as a result.

Functionally in this case with the current code it doesn't matter.  If you're stuck on this you're welcome to accept your suggestion and approve the PR.  Otherwise, I still think this makes sense as is and  will probably continue to use this style in the future.  There's been a lot of cases in the past where a choice that was binary took on more options and something zeroing out a list in the middle created bugs/made modifying the code harder.

Reply:
in that case shouldnt you do `to_poke += book.extras.maxiKeepers.gaugeRewardsInjectors.values()` instead? doesnt make any sense either, since it is the initialisation of the var imo

Reply:
Pick one and merge it. It's a style thing, the script is simple enough that it can easily be refactored later, you won't miss that.

High level reasoning

If patern is
list = []
maybe add to list
maybe add to list
maybe add to list
maybe add to list

Then I can add a maybe add to list anywhere in there and it works.  The init is clearly different from the adds.

If the pattern is

Maybe add to list or set to 0 if you can't
maybe add to list
maybe add to list
maybe add to list

It's less clear that one shouldn't put a maybe add to list above the add or set.  It's easy if you see a var init line that is simple and know to use that var after.